Overhaul "Learn Shiny" content to be Express focused (#68)
posit-dev/py-shiny-site: docs/express-introduction.qmd

1. **Decoupling of UI and server**. Decoupling forces you to separate concerns in such a way that it's easier refactor and reason about a large amount of code.
2. **Functional programming**. Express encourages a more imperative (less functional) style, which can be easier to write, but also harder to read. This is especially true when you need to create [reusable extractions](express-in-depth.qmd##expressify).
3. **More explicit UI**. Express has magical defaults that make things simpler, but can also mask what's actually going on behind-the-scenes, and offers less opportunities to customize behavior.

The next article illustrates how these downsides can be mitigated to some extent in Express with some advanced tooling.
However, if you feel yourself reaching for these tools, it might be time to switch to Core.
If you're already convinced that Shiny Core is right for your app, feel free to skip ahead to the [transitioning](express-to-core.qmd) article.

::: callout-note
### Shiny Core is a superset of Express

Shiny Core is a superset of Express, meaning that anything you can do in Express, you can also do in Core.
The reverse is not true, however.

> One example of this [Shiny Modules](modules.qmd), which are not currently supported in Express.

Reviewer: ```suggestion
For example [Shiny Modules](modules.qmd) are not currently supported in Express.
```
